prov/shm/src/smr_ep.c

```diff
@@ -381,14 +382,17 @@ int smr_iov_mmap_copy_in(struct smr_ep *ep, struct smr_region *peer_smr,
 		FI_WARN(&smr_prov, FI_LOG_EP_CTRL, "mmap error\n");
 		goto err2;
 	}
-
-	num = ofi_copy_from_iov(mapped_ptr, total_len, iov, count, 0);
-	if (num != total_len) {
-		FI_WARN(&smr_prov, FI_LOG_EP_CTRL, "copy from iov error\n");
-		goto err2;
+	if (op != ofi_op_read_req) {
+		num = ofi_copy_from_iov(mapped_ptr, total_len, iov, count, 0);
+		if (num != total_len) {
+			FI_WARN(&smr_prov, FI_LOG_EP_CTRL, "copy from iov error\n");
+			goto err2;
+		}
+		munmap(mapped_ptr, total_len);
+	} else {
+		*shm_ptr = mapped_ptr;
```

Comment:
This change makes the function confusing and the encapsulation weird.  The function is called 'copy_in', but now no copy is being performed, and the only use is to set a pointer using a value passed in by the caller.  So, the caller could just have easily set this.  And now the mmap is left, but the file is closed.

Reply:
We have combined copy_in() and format_mmap(). So shm file is opened and mmap'ed for all cases. 

> And now the mmap is left, but the file is closed.
I think it's valid.
As per http://man7.org/linux/man-pages/man2/mmap.2.html. 
After the mmap() call has returned, the file descriptor, fd, can be closed immediately without invalidating the mapping.